web/src/redux/githubID/githubID.reducer.ts

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable no-case-declarations */
+import GithubIDActionTypes from './user.types';
+// TODO (BILLY), try to use interface instead of any
+
+// interface GithubIdBaseInterface {
+//   user: object | null;
+//   inputKeyword: string | null;
+//   isLoading: boolean;
+//   fetched: boolean;
+//   notFound: boolean;
+// }
+
+// interface ProductOwnerGithubIDInterface extends GithubIdBaseInterface {
+//   productOwnerGithubID: {};
+// }
+// interface TechnicalLeadInitialState extends Array<GithubIdBaseInterface> {
+//   technicalLeads: {};
+// }
+// interface GithubIdInitialState extends ProductOwnerGithubIDInterface, TechnicalLeadInitialState {}
+
+const INITIAL_STATE: any = {
+  updatedProductOwner: {
+    githubUser: null,
+    inputKeyword: null,
+    isLoading: false,
+    everFetched: false,
+    notFound: false,
+  },
+  FirstUpdatedTechnicalLeads: {
+    githubUser: null,
+    inputKeyword: null,
+    isLoading: false,
+    everFetched: false,
+    notFound: false,
+  },
+  SecondUpdatedTechnicalLeads: {
+    githubUser: null,
+    inputKeyword: null,
+    isLoading: false,
+    everFetched: false,
+    notFound: false,
+  },
+};
```

Comment:
this could be better types. You could have a type for a 'user' for each type of persona.

Reply:
updated